### PR TITLE
Add modular build for raspi

### PR DESCRIPTION
**Pull request by andrewsavage1**
b/288332462

### CODE REVIEW

**Comment by kaidokert** [comment]
Somewhat tangential, but relevant topic: I've wanted to unify Raspi/Linux build code ( e.g. GN ) for a _long_ time. There's very little reason why they are different in the first place, it should all come from `starboard/common/linux` or such. 

The only refactoring issue is disabling certain media codecs at build-time that work fine on Desktop Linux but wouldn't build on Raspi, e.g go/cobalt-cl/207742 or such.

File: starboard/shared/ffmpeg/BUILD.gn
Comment (andrewsavage1) [reply starboard/shared/ffmpeg/BUILD.gn]: Put a hack I'm not happy with in place

**Comment by kaidokert** [comment]
Tagging @oxve to help review the relevant GN code

File: starboard/shared/ffmpeg/BUILD.gn
Comment (oxve) [inline starboard/shared/ffmpeg/BUILD.gn:25]: Should this be b/208910380?

File: .github/config/raspi-2-modular.json
Comment (oxve) [inline .github/config/raspi-2-modular.json:1]: Should this config be in raspi-2.json? The docker service and test config are identical.

File: starboard/shared/ffmpeg/BUILD.gn
Comment (andrewsavage1) [reply starboard/shared/ffmpeg/BUILD.gn:25]: It's kinda both—if your bug is fixed, this compiles, if the bug I have in the todo is fixed, it doesn't need to compile. We really want to fix both either way, but I have no hard pref on which is in the TODO

File: .github/config/raspi-2-modular.json
Comment (andrewsavage1) [reply .github/config/raspi-2-modular.json:1]: Ah you're probably right—I think I specifically separated out linux because we weren't ready to run black box tests on the modular build. Let me take another look

Comment (andrewsavage1) [reply .github/config/raspi-2-modular.json:1]: Actually no—I don't know of a way to set `modular: true` for just this platform as I currently do here: https://github.com/youtube/cobalt/pull/898/files#diff-34e47d5d549f4ecf4651da5d34a14c80c4a32b09eb0af3882a7c3eadfb31acc1R49. If you know of a cleaner way to do this I'm open to it

File: starboard/shared/ffmpeg/BUILD.gn
Comment (oxve) [reply starboard/shared/ffmpeg/BUILD.gn:25]: I opened b/292127297 yesterday which is also deals with the same stuff. @sideb0ard is investigating.

It sounded like this should all go away but I don't know what the timeline is.

File: .github/config/raspi-2-modular.json
Comment (oxve) [reply .github/config/raspi-2-modular.json:1]: Ah I see. Yeah, that won't work. I think it would work if you move that attribute to the `includes` section like we do for other build arguments.

I want to start a discussion about refactoring some of the pipeline configs so we can make a definite decision about how we handle build/test inputs. Given that you should decide how you want to do this now.

File: starboard/build/toolchain/BUILD.gn
Comment (andrewsavage1) [inline starboard/build/toolchain/BUILD.gn:27]: Looking for a clean way to configure this

Comment (y4vor) [reply starboard/build/toolchain/BUILD.gn:27]: We can try to  use the same pattern as the one we use for setting the path:

clang_toolchain("x64") {
  clang_base_path = clang_base_path
  ld = custom_ld
}

**Review by y4vor — APPROVED** [review]

File: starboard/build/toolchain/BUILD.gn
Comment (niranjanyardi) [inline starboard/build/toolchain/BUILD.gn:17]: this should ideally go,

same file will have clang3-6 , clang 6-3 , all of these cant be included here
might be good to define a variable like yavor mentioned :  native_ld  which is then set for all 
platforms

File: starboard/build/toolchain/BUILD.gn
Comment (niranjanyardi) [reply starboard/build/toolchain/BUILD.gn:27]: https://github.com/youtube/cobalt/pull/898/files#r1291615262

File: starboard/build/toolchain/BUILD.gn
Comment (niranjanyardi) [reply starboard/build/toolchain/BUILD.gn:17]: maybe in the paltform's configuration.gni and then gets imported here

Comment (niranjanyardi) [reply starboard/build/toolchain/BUILD.gn:17]: this  file shouldnt know about clang3-6, raspi, ps4/5 or any other platform specific gni at this level i think

File: starboard/evergreen/shared/platform_configuration/BUILD.gn
Comment (niranjanyardi) [inline starboard/evergreen/shared/platform_configuration/BUILD.gn:26]: -shared flag will be needed by modular 

ideally should be refactored out in a modular folder in build/config/ so that other platforms can use this and some of the other common compiler flags : can be done in a later cl

File: starboard/raspi/shared/launcher.py
Comment (niranjanyardi) [inline starboard/raspi/shared/launcher.py:146]: can this be abstracted out for all launchers ?

Okay to do it in a separate cl

File: starboard/raspi/shared/launcher.py
Comment (niranjanyardi) [inline starboard/raspi/shared/launcher.py:153]: the whole launcher thing lgtm if tests pass for all- modular, evergreen, monolithic

File: starboard/raspi/shared/platform_configuration/BUILD.gn
Comment (niranjanyardi) [inline starboard/raspi/shared/platform_configuration/BUILD.gn:34]: i think we need to clean up all these unresolved symbols in alter cl - this apaprently is hard to do for ps4

@kaidokert  can you ptal

Comment (niranjanyardi) [reply starboard/raspi/shared/platform_configuration/BUILD.gn:34]: this shouldnt block this PR IMO , so feel free to resolve

File: starboard/raspi/shared/platform_configuration/configuration.gni
Comment (niranjanyardi) [inline starboard/raspi/shared/platform_configuration/configuration.gni:40]: dunno if @y4vor has comments.
Again feel free to resolve and submit

**Review by niranjanyardi — APPROVED** [review]

File: starboard/raspi/shared/platform_configuration/configuration.gni
Comment (andrewsavage1) [reply starboard/raspi/shared/platform_configuration/configuration.gni:40]: This hasn't changed, the diff is just a little off

File: starboard/BUILD.gn
Comment (oxve) [inline starboard/BUILD.gn:54]: A little confused by this. Why do you have to add the install target for player_filter_tests here? And if you really do should it replace the line above instead?

File: starboard/raspi/2/starboard_loader.cc
Comment (oxve) [inline starboard/raspi/2/starboard_loader.cc:1]: Just out of curiosity, where does this file get pulled in? Is that already set up?

Comment (andrewsavage1) [reply starboard/raspi/2/starboard_loader.cc:1]: Here: https://github.com/youtube/cobalt/pull/898/files/20fe4c3d8eeffa86d543a6ca1eb56291bd7b9946#diff-edd3910ce63e3202b04f07359958177d3208852d1e60eaecb6c6c25ec29b7a9dR502

File: starboard/BUILD.gn
Comment (andrewsavage1) [reply starboard/BUILD.gn:54]: It's a quirk of GN—I believe if nothing depends on a target built with a toolchain other than the default, it will not be in the ninja database.

I could probably just replace player_filter_tests above, but let's do that in a follow up as I have to make sure it isn't affected by b/296884941

**Review by oxve — APPROVED** [review]

**Comment by andrewsavage1** [comment]
Going to submit this—raspi tests for both modular and monolithic all pass locally, CI is just having issues with them

**Pull request merged**
